README.md

```diff
@@ -53,6 +53,64 @@ You'll need a [browser Buffer polyfill](https://www.npmjs.com/package/browserify
 
 If you want to support node & browsers with the same code, you can use the latter `await` form here everywhere (since awaiting the fixed value in node just returns the value as-is).
 
+
+## Vite and Rollup
+
+For usage in Vite, you have to use the `vite-plugin-wasm` and `static-files` plugins. The config looks like this:
+
+```javascript
+import { defineConfig } from "vite";
+import wasm from "vite-plugin-wasm";
+import { wasm as rollupWasm } from "@rollup/plugin-wasm";
+import { viteStaticCopy } from "vite-plugin-static-copy";
+
+// https://vitejs.dev/config/
+export default defineConfig({
+    plugins: [
+        viteStaticCopy({
+            targets: [
+                {
+                    src: require.resolve("brotli-wasm/pkg.web/brotli_wasm_bg.wasm"),
+                    dest: "."
+                }
+            ]
+        }),
+        wasm(),
+        topLevelAwait()
+    ],
+    build: {
+        minify: false,
+        target: ["esnext"],
+        sourcemap: true,
+        rollupOptions: {
+            treeshake: false,
+            plugins: [rollupWasm()]
+        },
+        ssr: false
+    }
+});
+```
+
+and you can call it in your code like this:
+
+```typescript
+import init, { decompress } from "brotli-wasm/pkg.web/brotli_wasm";
```

Comment:
It would be nice to use conditional exports but I'm not entirely sure it's possible.

The issue is that both `pkg.bundler` and `pkg.web` would be suitable for a `web` environment, and can be called from either an esm or cjs scope. The condition to diverge on would be the bundler being used, which doesn't seem to be something that we can filter on in [conditional exports](https://nodejs.org/api/packages.html#conditional-exports).

https://github.com/rollup/plugins/issues/362

Rollup support conditional exports, but cannot differentiate between different browser versions. That's to say that it'll always just pickup what's being used in `browser`. We could define a community condition and open a PR with rollup to support that first (like a top level `rollup` key).